### ISSUE TITLE
Regression: Use the function form of "use strict".

### ISSUE THREAD

**Issue by Raynos** [jshint/jshint#2661]
We depend on jshint with `^` and our travis builds started failing complaining about use function form of "use strict".

We have the following set:

```
{
    "strict": false,
    "globalstrict": true,
    ...
}
```

This is a breaking change and will probably break other people as well.

**Comment by Raynos** [comment]
looks like 8de8247 is a regression

```
-        if (state.isStrict()) {
-          if (!state.option.globalstrict) {
+        if (state.directive["use strict"]) {
+          if (state.option.strict !== "global") {
```

**Comment by Raynos** [comment]
There are more regressions:

```
> tchannel@2.8.1 lint /home/raynos/uber/tchannel/node
> jshint .

benchmarks/bench_server.js: line 114, col 9, Reassignment of 'onPing', which is is a function. Use 'var' or 'let' to declare bindings that may change.
benchmarks/bench_server.js: line 117, col 9, Reassignment of 'onSet', which is is a function. Use 'var' or 'let' to declare bindings that may change.
benchmarks/bench_server.js: line 120, col 9, Reassignment of 'onGet', which is is a function. Use 'var' or 'let' to declare bindings that may change.

test/hyperbahn/advertise.js: line 88, col 13, 'client' was used before it was defined.
```

Please dont publish a new minor version with 3 months worth of commits -.-

**Comment by lukeapage** [comment]
@Raynos please raise the other "regressions" in a different issue with code samples. it is possible some of those new warnings are rule violations that were not previously detected ratger than regressions.

**Comment by lukeapage** [comment]
possible duplicate of https://github.com/jshint/jshint/issues/2657

**Comment by nightpool** [comment]
I think this is the inverse of #2657—code that wasn't strict, beforehand, is now getting interpreted as strict. 

I think. @0xazure does that make sense?

**Comment by lukeapage** [comment]
@Raynos We have fixed the reported regression with use strict.
Investigating your other regressions, looking at https://github.com/uber/tchannel/blob/master/node/ - the advertise.js isn't in the repo so I cannot comment on that, but the new errors in bench_server are as I suggested, warnings that were not given previously.

E.g. in 2.8.0

```
function a() {
}
a = 1; // warns
b = 1; // does not warn
function b() {
}
```
 in 2.9.0
```
function a() {
}
a = 1; // warns
b = 1; // warns
function b() {
}
```

it is a difficult line between a regression and a bugfix but in this case we decided it was a bugfix and should now warn.

@jugglinmike are you happy with the above still or do you think we should not fix it till a major release?

**Comment by Raynos** [comment]
@lukeapage I think its fine :)

**Comment by lukeapage** [comment]
Lets keep this open till the use strict fix is merged.

**Comment by nicolo-ribaudo** [comment]
This can be closed

**Comment by totallymike** [comment]
Is this still closable?  It looks like the commits referencing this issue weren't merged into master.

**Comment by nicolo-ribaudo** [comment]
04b43d22755a766311f3ed12a2e6ce51d6ac4a3a

**Comment by totallymike** [comment]
Awesome.  :+1: to closing then

**Comment by pragmadash** [comment]
The same issue is reproduced for me using version 2.9.1.

My config:

```javascript
{
	"maxerr": 15,

	"nonew": true,
	"proto": true,
	"maxparams": 5,
	"maxdepth": 3,
	"maxstatements": 50,
	"maxcomplexity": 10,

	"globalstrict": true,

	"browser": false,
	"node": true,

	"globals": {
		"Promise": true,
		"window": true,
		"describe": true,
		"it": true
	}
}
```

Console output:

```
browser/DocumentRenderer.js: line 867, col 1, Incompatible values for the 'strict' and 'globalstrict' linting options. (65% scanned).

lib/streams/ComponentReadable.js: line 150, col 1, Incompatible values for the 'strict' and 'globalstrict' linting options. (29% scanned).

lib/tokenizers/HTMLTagTokenizer.js: line 107, col 1, Incompatible values for the 'strict' and 'globalstrict' linting options. (21% scanned).

test/lib/finders/InjectionFinder.js: line 57, col 21, Incompatible values for the 'strict' and 'globalstrict' linting options. (64% scanned).

4 errors
```

The content on the lines is:

browser/DocumentRenderer.js: line 867
lib/streams/ComponentReadable.js: line 150, col 1
lib/tokenizers/HTMLTagTokenizer.js: line 107
```javascript
/*jshint maxcomplexity:false */
```

test/lib/finders/InjectionFinder.js: line 57
```javascript
/*jshint evil:true*/
```

Please do not release breakable change as a minor release.

**Comment by nicolo-ribaudo** [comment]
@pragmadash The issue you pointed out has been fixed in #2843

**Comment by pragmadash** [comment]
@nicolo-ribaudo thank you for replying. Waiting for a release with that fix. Hope it won't take long, because all our build are failing right now.

**Comment by jugglinmike** [comment]
The fix has been released in JSHint 2.9.2: http://jshint.com/blog/2016-04-19/release-2-9-2/
